teps/0146-parameters-in-script.md

```diff
@@ -0,0 +1,605 @@
+  steps:
+  - name: foo
+    image: myimage
+    env:
+    - name: PARAM_URL
+      value: $(params.url)
+    script: |
+      git clone ${PARAM_URL}
+```
+### Discussion
+This approach mutates the specification, but it is something we're already doing
+with implicit `Parameters` per [TEP-0023][tep-0023].  This approach removes the insecure
+default behaviour but in doing so unfortunately loses backwards-compatibility requiring users
+to modify their `script` yaml (eg: `s/$(params.url)/${$(params.url)}` for the `shell` case).
+This approach also has a difficult deprecation/onboarding story for users as there is no built in
+alternative and changing defaults is a hard cutoff for them (eg: they are broken with a new release as 
+the defaults change) vs for example an alternative method for doing this and a timeline for deprecating the 
+default
+
+<!-- ### Notes and Caveats -->
+
+<!--
+(optional)
+
+Go in to as much detail as necessary here.
+- What are the caveats to the proposal?
+- What are some important details that didn't come across above?
+- What are the core concepts and how do they relate?
+-->
+
+## Design Details
+See above, more information is WIP - will update 10/3
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+
+If it's helpful to include workflow diagrams or any other related images,
+add them under "/teps/images/". It's upto the TEP author to choose the name
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+
+## Design Evaluation
+See above, more information is WIP - will update 10/3
+<!--
+How does this proposal affect the api conventions, reusability, simplicity, flexibility
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+### Reusability
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#reusability
+
+- Are there existing features related to the proposed features? Were the existing features reused?
+- Is the problem being solved an authoring-time or runtime-concern? Is the proposed feature at the appropriate level
+authoring or runtime?
+-->
+- Are there existing features related to the proposed features? Were the existing features reused?
+  - The solution outlined here is similar to the current `.path` field supported in `$(workspaces.<name>.path)`.
+  - A related feature can be used in combination with this TEP is [[TEP-0144] Param Enum](https://github.com/tektoncd/community/pull/1072)
+
+### Simplicity
+
+- How does this proposal affect the user experience?
+  - This proposal draws upon usage of a suffix (`.env`) on a param similar to how `.path` is supported 
```

Comment:
I guess this `.env` syntax is the proposed solution that will go into the design details?

Reply:
Oh, it's below in the alternatives, never mind.

Reply:
I have clarified in the document now that the preferred solution would be the `.env` solution which is explicit as any implicit change is backwards compat.  This (explicitly defining to use the env var interpolation) was the intended preferred solution so I've now changed the document to make this more clear.  Marking this resolved, lmk if there is more to discuss here.